File: 04-JavaScript-ExpressJS-Overview/Calculator/models/Calculator.js
function Calculator(leftOperand, rightOperand, operator) {
    this.leftOperand = leftOperand;
    this.rightOperand = rightOperand;
    this.operator = operator;

    this.calculateResult = function () {
        let result = 0;

        switch (this.operator){
            case '+':
                result = this.leftOperand + this.rightOperand;
                break;
            case '-':
                result = this.leftOperand - this.rightOperand;
                break;
            case '*':
                result = this.leftOperand * this.rightOperand;
                break;
            case '/':
                result = this.leftOperand / this.rightOperand;
                break;
            case 'mod':
                result = this.leftOperand % this.rightOperand;
                break;
            case 'pow':
                result = Math.pow(this.leftOperand, this.rightOperand);
                break;
            case 'What is X% of Y':
                result = this.leftOperand / 100 * this.rightOperand;
                break;
            case 'X is what % of Y':
                result = this.leftOperand / this.rightOperand * 100;
                break;
            case '√X':
                result = Math.sqrt(this.leftOperand);
                break;
            case 'X!':
                function calcFactorial(n) {
                    let factorial = 1;
                    while (n> 1)
                    {
                        factorial *= n;
                        n--;
                    }

                    return factorial;
                }

                result = calcFactorial(this.leftOperand);
                break;
        }

        return result;
    }
}

module.exports = Calculator;

File: 10-Java-Basic-Web/Calculator/src/main/java/com/softuni/entity/Calculator.java
package com.softuni.entity;

public class Calculator {
    private double leftOperand;
    private double rightOperand;
    private String operator;

    public Calculator(double leftOperand, double rightOperand, String operator) {
        this.leftOperand = leftOperand;
        this.rightOperand = rightOperand;
        this.operator = operator;
    }

    public double getLeftOperand() {
        return leftOperand;
    }

    public void setLeftOperand(double leftOperand) {
        this.leftOperand = leftOperand;
    }

    public double getRightOperand() {
        return rightOperand;
    }

    public void setRightOperand(double rightOperand) {
        this.rightOperand = rightOperand;
    }

    public String getOperator() {
        return operator;
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public double CalculateResult() {
        double result = 0;

        switch (this.operator) {
            case "+":
                result = this.getLeftOperand() + this.getRightOperand();
                break;
            case "-":
                result = this.getLeftOperand() - this.getRightOperand();
                break;
            case "*":
                result = this.getLeftOperand() * this.getRightOperand();
                break;
            case "/":
                result = this.getLeftOperand() / this.getRightOperand();
                break;
            case "mod":
                result = this.getLeftOperand() % this.getRightOperand();
                break;
            case "pow":
                result = Math.pow(this.getLeftOperand(), this.getRightOperand());
                break;
            case "And":
                result = (int)this.getLeftOperand() & (int)this.getRightOperand();
                break;
            case "Or":
                result = (int)this.getLeftOperand() | (int)this.getRightOperand();
                break;
            case "Xor":
                result = (int)this.getLeftOperand() ^ (int)this.getRightOperand();
                break;
            case "Not":
                result = ~ (int)this.getLeftOperand();
                break;
            case "ShiftLeft":
                result = (int)this.getLeftOperand() << (int)this.getRightOperand();
                break;
            case "ShiftRight":
                result = (int)this.getLeftOperand() >> (int)this.getRightOperand();
                break;
            case "What is X% of Y":
                result = this.getLeftOperand() / 100 * this.getRightOperand();
                break;
            case "X is what % of Y":
                if (this.getRightOperand() != 0) {
                    result = this.getLeftOperand() / this.getRightOperand() * 100;
                }
                break;
            case "√X":
                result = Math.sqrt(this.getLeftOperand());
                break;
            case "X!":
                result = calcFactorial(this.getLeftOperand());
                break;
        }

        return result;
    }

    private double calcFactorial(double n) {
        double factorial = 1.0;

        while (n > 1.0)
        {
            factorial *= n;
            n--;
        }

        return factorial;
    }
}